src/write/file.rs

```diff
@@ -119,8 +126,13 @@ impl<W: Write> FileWriter<W> {
         Ok(())
     }
 
-    /// Writes the footer of the parquet file. Returns the total size of the file.
-    pub fn end(&mut self, key_value_metadata: Option<Vec<KeyValue>>) -> Result<u64> {
+    /// Writes the footer of the parquet file. Returns the total size of the file and the
+    /// underlying writer.
+    pub fn end(mut self, key_value_metadata: Option<Vec<KeyValue>>) -> Result<(u64, W)> {
+        if self.offset == 0 {
+            self.start()?;
+        }
+
```

Comment:
Consuming the FileWriter synchronises the API with the async version, and also prevents inadvertently calling end twice